packages/rest/src/lib/handlers/Shared.ts

```diff
@@ -137,15 +139,29 @@ export async function handleErrors(
 	} else {
 		// Handle possible malformed requests
 		if (status >= 400 && status < 500) {
+			// The request will not succeed for some reason, parse the error returned from the api
+			const data = (await parseResponse(res)) as DiscordErrorData | OAuthErrorData;
+			const isDiscordError = 'code' in data;
+
 			// If we receive this status code, it means the token we had is no longer valid.
 			if (status === 401 && requestData.auth === true) {
+				if (isDiscordError && data.code !== 0 && !authFalseWarningEmitted) {
+					const errorText = `Encountered HTTP 401 with error ${data.code}: ${data.message}. Your token will be removed from this REST instance. If you are using @discordjs/rest directly, consider adding 'auth: false' to the request. Open an issue with your library if not.`;
+					// Use emitWarning if possible, probably not available in edge / web
+					if (typeof process?.emitWarning === 'function') {
```

Comment:
Seems more likely that `process` wouldn't be declared at all rather than being nullish

```suggestion
					if (typeof process !== 'undefined' && typeof process.emitWarning === 'function') {
```

Reply:
Should this be `globalThis`?